webrender/src/resource_cache.rs

```diff
@@ -764,6 +764,24 @@ impl ResourceCache {
         debug_assert_eq!(self.state, State::QueryResources);
         self.state = State::Idle;
     }
+
+    pub fn clear_namespace(&mut self, namespace: IdNamespace) {
+        //TODO: use `retain` when we are on Rust-1.18
```

Comment:
These would eventually expire on their own, right? But yeah sounds like it would be a good thing to do as well, and also remove items from the texture cache when removing image and font templates.